tools/rpctypegen/src/lib.rs

```diff
@@ -0,0 +1,165 @@
+extern crate proc_macro;
+extern crate proc_macro2;
+
+use proc_macro::TokenStream;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::cell::RefCell;
+use std::collections::BTreeMap;
+use syn::{
+    parse_macro_input, Data, DataEnum, DataStruct, DeriveInput, Fields, FieldsNamed, FieldsUnnamed,
+    Lit, Meta, MetaNameValue,
+};
+
+fn merge(a: &mut Value, b: &Value) {
+    match (a, b) {
+        (&mut Value::Object(ref mut a), &Value::Object(ref b)) => {
+            for (k, v) in b {
+                merge(a.entry(k.clone()).or_insert(Value::Null), v);
+            }
+        }
+        (a, b) => {
+            *a = b.clone();
+        }
+    }
+}
+
+#[derive(Default, Debug, Deserialize, Serialize)]
+struct Schema {
+    pub schema: BTreeMap<String, ErrorType>,
+}
+
+impl Drop for Schema {
```

Comment:
I don't understand why it needs to use files within `target` at `Drop`. That's bad design.

Reply:
As discussed, we should remove this hack from the common flow. So either hide behind some `[cfg(feature)]` or move to a separate binary.

Reply:
take a look https://github.com/nearprotocol/nearcore/pull/1854/commits/66331ed8987e7d23150255e4435fab7a161f943a

Reply:
I assume the proc macro is still enabled, so the schema will still be constructed. @nearmax @frol can you also take a look?

My concern is the schema aggregation code will still be executed even if the feature is off.

Reply:
I don’t think it’s a bad thing. It’s good to be sure that annotations are valid even  if we don’t do any generation.

Reply:
I guess @nearmax is about create an issue about another approach..